### ISSUE TITLE
UX: It is too easy to send to a legacy format address

### ISSUE THREAD

**Issue by zander** [Electron-Cash/Electron-Cash#2233]
In the receive tab you can paste a legacy address and due to a silly bug it is still very high risk that people will send money to such an address.

To reproduce;

go to the send tab, paste a 1* type address. Set '1' for the fiat amount and
Press 'preview'. 

You are presented with a dialog that nicely warns you about the address being legacy. At this time I think the 
text is much too mild as "deprecated" is far too weak a statement. But thats out of scope for this issue.

Focus on the bottom of the dialog, notice that there is no "cancel". Just "show details" and "ok".
We can do nothing but press "ok" here and that leads us to the next dialog that shows the preview. With the 'sign' and 'broadcast' buttons.

What this basically means is that the nag dialog that is added has no effect. People that don't read dialogs (that is most of them) will not notice any change in flow and will just send the money to the wrong address anyway.

I strongly suggest;

* Adding a "cancel" button to the warning dialog.
* Wording it much stronger. People should know what they are doing, eveyrone else is going to lose money.
* Remove the "never ask again" checkbox.

In short, treat the legacy address format not like a legacy format, treat it like an incompatible BTC address, unless the user is very insistant on being competent enough.

**Comment by slavomirtravica** [comment]
@GitCash send .002 bch to @zander

**Comment by monsterbitar** [comment]
I'd go as far as to move support for Base58check addresses to be an off-by-default setting you have to enable if you want to use it.

**Comment by cculianu** [comment]
Yeah I'm down with that @monsterbitar  -- however I noticed many Chinese users still give out / use Legacy addresses on BCH, oddly. Not sure why that is. Maybe some Chinese multi-coin wallets do that for BCH that's why?  

Hmm.

We could have it off by default but have the pop-up "ask you to enable" right there in the popup (with a "never show me again" checkbox if you say No)... maybe?

**Comment by seanPhill** [comment]
Can you (or the wallet) just explain in what scenario someone might lose their BCH just by sending to the legacy format of the address?
What is this "silly bug"? I've used legacy addresses without problem when withdrawing from exchanges that did not support the CashAddress.

But what I'm really looking for (and not finding) while searching these issues is whether a "simpleledger" address is fine for receiving BCH, and why converting one to the "bitcoincash" address format yields a different address.
Clearly it is useful to identify an address as being in a wallet capable of handling SLP tokens, but it makes it unclear whether it can also be used for ordinary Bitcoin Cash. (Sorry for expanding the scope of the original question, but I couldn't find another relevant context.)

**Comment by zander** [comment]
Maybe the original issue is that its not clear what the cost or problem is that is being solved here.

We invented the cashaddress standard to make it near impossible for normal people to confuse coins.
When everyone uses cashaddress you don't mistakenly send BTC to a BCH payment request and vice versa.

The fact that EC so nicely falls back into supporting the BTC address format undo's this protection.
Now an exchange or merchant asking for a BTC address for a BCH deposit may get BTC deposited on that address.
vice versa, a BTC payment request can be fulfilled in EC and also there funds are being lost.

Secondary issue is that people like Joel and other community members are having problems supporting people as we always have to check assumptions and check chains.

So, again, the entire point of having a different address format is undermined by wallets making it easy to sidestep the protection it gives.

**Comment by zander** [comment]
In short, supporting the BTC format is Ok, but it should be a very clear and conscious choice of the user which understands the effects and make sure they are not losing money.

People losing their money cross-chain is making people leave BCH and/or crypto, going back to the much safer fiat or btc options.

Don't chase away users. Protect them.

**Comment by cculianu** [comment]
![telegram-cloud-photo-size-4-5890905851487698694-y](https://github.com/Electron-Cash/Electron-Cash/assets/266627/e98a3876-c81b-496b-a5ed-0bd4e2b0f66e)

**Comment by cculianu** [comment]
Ok, latest commit 3edd940 handles this better. This is on macOS:

<img width="510" alt="Screen Shot 2023-10-24 at 7 46 26 PM" src="https://github.com/Electron-Cash/Electron-Cash/assets/266627/b377a15b-f8ee-40c3-8a73-82b31202f5e2">
